pint_pandas/pint_array.py

```diff
@@ -1033,4 +1033,9 @@ def is_pint_type(obj):
         return False
 
 
-compat.upcast_types.append(PintArray)
+try:
+    # for pint < 0.21 we need to explicitly register
+    compat.upcast_types.append(PintArray)
+except AttributeError:
+    # for pint >= 0.21 we don't have to register explicitly
```

Comment:
Due to https://github.com/hgrecco/pint/pull/1777 it would be good to add:
```suggestion
    # for pint = 0.21 we need to add the full name, which is to be added in pint > 0.21
    compat.upcast_types_map.setdefault("pint_pandas.pint_array.PintArray", None)
```